gitlab/provider.go

```diff
@@ -116,14 +121,18 @@ func init() {
 	}
 }
 
-func providerConfigure(d *schema.ResourceData) (interface{}, error) {
+func providerConfigure(p *schema.Provider, d *schema.ResourceData) (interface{}, error) {
 	config := Config{
 		Token:      d.Get("token").(string),
 		BaseURL:    d.Get("base_url").(string),
 		CACertFile: d.Get("cacert_file").(string),
 		Insecure:   d.Get("insecure").(bool),
 		ClientCert: d.Get("client_cert").(string),
 		ClientKey:  d.Get("client_key").(string),
+
+		// NOTE: httpclient.TerraformUserAgent is deprecated and removed in Terraform SDK v2
+		// After upgrading the SDK to v2 replace with p.UserAgent()
+		TerraformUserAgent: httpclient.TerraformUserAgent(p.TerraformVersion),
 	}
 
```

Comment:
client := provider.Client()
client.UserAgent = httpclient.TerraformUserAgent(p.TerraformVersion)
return client

Reply:
Done. Personally I am not a fan of the `providerConfigure` function customizing the `client` as that seems to be the responsibility of the `Client()` function but the code is indeed simpler.